superset-frontend/src/explore/components/ControlPanelsContainer.test.tsx

```diff
@@ -144,4 +144,41 @@ describe('ControlPanelsContainer', () => {
       await screen.findAllByTestId('collapsible-control-panel-header'),
     ).toHaveLength(2);
   });
+
+  test('hide ControlPanelSections when its visibility is false', async () => {
```

Comment:
You're also testing if the panel is displayed when the visibility is `true`.
```suggestion
  test('visibility of panels is correctly applied', async () => {
```